[libc] pthread_setspecific has wrong fn signature in generated header

impl looks correct, generated header looks incorrect:
```c
void * pthread_setspecific(pthread_key_t, const void *) __NOEXCEPT;
```
fix libc/include/pthread.yaml for `pthread_setspecific` should return `int`, not `void*`.

Otherwise the following error is observed when building libcxxabi against llvm-libc:
```
/llvm-project-main/build/include/c++/v1/__thread/support/pthread.h:216:10: error: cannot initialize return object of type 'int' with an rvalue of type 'void *'
  216 |   return pthread_setspecific(__key, __p);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/libc/include/pthread.yaml
+++ b/libc/include/pthread.yaml
@@ -402,7 +402,7 @@ functions:
   - name: pthread_setspecific
     standards:
       - POSIX
-    return_type: void *
+    return_type: int
     arguments:
       - type: pthread_key_t
       - type: const void *

